Adding change notes.md

```diff
@@ -0,0 +1,69 @@
+# Adding change notes for query and library changes
+
+Each CodeQL query pack or library pack has its own change log to track how that pack changes with each release. Any non-trivial, user-visible change to a query or library should add a change note to the affected pack. This document describes how to do that.
```

Comment:
We should have a change note validation step as part of CI. It should just ensure the change notes are well-formed. Not blocking for this change, though.

Reply:
Yeah, that's the plan. Probably just a switch to `codeql pack changelog` or something.